compiler/rustc_data_structures/src/graph/vec_graph/mod.rs

```diff
@@ -20,7 +20,7 @@ pub struct VecGraph<N: Idx> {
 impl<N: Idx> VecGraph<N> {
     pub fn new(num_nodes: usize, mut edge_pairs: Vec<(N, N)>) -> Self {
         // Sort the edges by the source -- this is important.
-        edge_pairs.sort();
+        edge_pairs.sort_by_key(|&edge_pairs| (edge_pairs.0.index(), edge_pairs.1.index()));
```

Comment:
This seems a little concerning - I would rather expect that this data structure/method would require that the index type implements Ord. (Some types which are comparable, e.g. mir block indices, should continue to implement Ord, even as we remove it from DefId). Otherwise we're just hiding the actual Ord dependence which doesn't really help, imo.

Reply:
How does this look? https://github.com/rust-lang/rust/pull/90408/commits/44e023a64901a47348284b67b2c32c3817c1beb3